src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -0,0 +1,228 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import feign.FeignException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException.BadRequest;
+import org.springframework.web.client.HttpClientErrorException.UnprocessableEntity;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.ServiceResponseParser;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.caseupdate.CaseUpdateClient;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.caseupdate.model.response.CaseUpdateDetails;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.caseupdate.model.response.SuccessfulUpdateResponse;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.model.request.ExceptionRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.model.response.ClientServiceErrorResponse;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.config.ServiceConfigItem;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.callback.CallbackException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.callback.ProcessResult;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+
+@Service
+public class CcdCaseUpdater {
+    private static final Logger log = LoggerFactory.getLogger(CcdCaseUpdater.class);
+
+    private final CaseUpdateClient caseUpdateClient;
+    private final ServiceResponseParser serviceResponseParser;
+    private final AuthTokenGenerator s2sTokenGenerator;
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final ExceptionRecordFinalizer exceptionRecordFinalizer;
+
+    public CcdCaseUpdater(
+        CaseUpdateClient caseUpdateClient,
+        ServiceResponseParser serviceResponseParser,
+        AuthTokenGenerator s2sTokenGenerator,
+        CoreCaseDataApi coreCaseDataApi,
+        ExceptionRecordFinalizer exceptionRecordFinalizer
+    ) {
+        this.caseUpdateClient = caseUpdateClient;
+        this.serviceResponseParser = serviceResponseParser;
+        this.s2sTokenGenerator = s2sTokenGenerator;
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.exceptionRecordFinalizer = exceptionRecordFinalizer;
+    }
+
+    @SuppressWarnings({"squid:S2139", "unchecked"}) // squid for exception handle + logging
```

Comment:
Why suppress `squid:S2139`? We shouldn't log the exception and then rethrow it, because that would result in two error log entries.